fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/service/JdbcTenantDetailsService.java

```diff
@@ -49,7 +49,9 @@ public JdbcTenantDetailsService(@Qualifier("hikariTenantDataSource") final DataS
 
     private static final class TenantMapper implements RowMapper<FineractPlatformTenant> {
 
-        private final StringBuilder sqlBuilder = new StringBuilder("t.id, ts.id as connectionId , ")//
+        private final String tenantIdentifier;
```

Comment:
This is starting to break the RowMapper pattern a bit... IMHO the purpose of RowMapper is to map one record of ResultSet to corresponding DTO. It should not care about what was the query used to get the result set - i.e. you should be able to reuse the same RowMapper in anything that queries a specific table. That's why the schema() method should return just the schema of the object, not the where condition as well.

Was there a reason why the where condition was moved to the schema()? Why not stick to the pattern and have the schema() return the schema for the object, and the calling method specify the where condition?

Reply:
Noted on pattern breakage. 
The change was informed by the need to keep the query construct constant and parameterize external input to the SQL query. 

We cannot add ? for `select  " + rm.schema()`. PreparedStatement will return `SELECT 'the string value of rm.schema'`
With a simple example such as `SELECT firstname FROM m_appuser;`
`SELECT ?;` parameterizing `firstname FROM m_appuser`  will  translate into a `SELECT 'firstname FROM m_appuser';` which returns resultSet value of  'firstname FROM m_appuser' 

It depends on PreparedStatement set# methods and Since we cannot user parameter queries for  `SELECT ? FROM ?; ` or SELECT ? for now I suggest. 
1. Leave this concatenation the way it is currently : `final String sql = "select  " + rm.schema();`  - it is from a 'trusted' source
2. In future - where Parameterized Queries is not feasible to mitigate SQL Injection, try ESAPI `final String sql = "select  " + EsapiUtils.encodeForMySQL(rm.schema());` 
3. I am trying ESAPI in PR #1687 but still in the kitchen sink and haven't tested things there

Reply:
I think leaving the concatenation as-is would be OK - as you say, we are not concatenating any user input / parameters but simply the hardcoded list of columns returned by a function. I don't think that introduces any additional security etc risks.